action.yml

```diff
@@ -38,13 +41,40 @@ runs:
       with:
         python-version: "3.12"
 
+    - id: set-binary
+      env:
+        TERRAFORM_BINARY: ${{ inputs.terraform_binary }}
+      run: |
+        # sanitise input
+        if [[ "$TERRAFORM_BINARY" == "tofu" ]] ; then
+          printf 'TF=%s\n' "tofu" >> "$GITHUB_ENV"
+          printf 'TF_MANAGER=%s\n' "tofuenv" >> "$GITHUB_ENV"
+        elif [[ "$TERRAFORM_BINARY" == "terraform" ]] ; then
+          printf 'TF=%s\n' "terraform" >> "$GITHUB_ENV"
+          printf 'TF_MANAGER=%s\n' "tfenv" >> "$GITHUB_ENV"
+        else
+          1>&2 printf 'Unsupported terraform binary %s\n' "$TERRAFORM_BINARY"
+          exit 1
+        fi
+      shell: bash
+
     - name: Setup tfenv
       if: ${{ inputs.terraform_version != 'system' }}
       run: |
-        if [[ ! -d ~/.tfenv ]]; then
-          git clone -b v3.0.0 https://github.com/tfutils/tfenv.git ~/.tfenv
+        if [[ "$TF" == "tofu" ]] ; then
+          if [[ ! -d ~/.tfenv ]]; then
+            git clone -b v1.0.6 https://github.com/tofuutils/tofuenv.git ~/.tofuenv
+          fi
```

Comment:
I think we should use tenv instead that supports both TF and Tofu
https://github.com/tfutils/tfenv

But feel free to left it be as well and we can do that in separate PR.

Reply:
The tfenv maintainers did not add support for tofu and instead asked the community to handle it separately, so tofuenv was forked from it and can manage opentofu versions. It's what I am using for my projects.

Reply:
Just to say it works mostly closely to how tfenv works.

Reply:
Sorry, wrong URL right there. I was talking about tenv 😅 
https://github.com/tofuutils/tenv

Reply:
Yes, I have seen this one too - the maintainer spammed anyone using tofuenv with a link to it :) It's a go binary and it's probably fine to use too, but needs a bit of different setup in the action.

If we want to do that, we should switch to using it in another PR I think and and then rebase this onto instead of conflating the two changes.

Reply:
example: https://github.com/SlackBuildsOrg/slackbuilds/pull/4430#issuecomment-1916573569

Reply:
Yes, definitely we (me or @hanscg ) can add it in another PR 👍

Reply:
And also, it's actually official from the tofu team and they recommend it instead of tfenv or tofuenv
https://github.com/tofuutils/tofuenv?tab=readme-ov-file#important-notice
> we are finally ready to announce a successor for tfenv and tofuenv: [tenv ](https://github.com/tofuutils/tenv)🚀 written in Golang

Reply:
> Yes, definitely we (me or @hanscg ) can add it in another PR 👍

I can prepare this as well. I'll open up another PR for review of that. Thanks.

Reply:
https://github.com/HENNGE/terraform-check/pull/38